Public/Src/Engine/Dll/Distribution/RemoteWorker.cs

```diff
@@ -1032,37 +1035,10 @@ private void Stop()
                     break;
                 }
 
-                var context = m_masterService.Environment.Context;
-
+                // Here, we fail the pips which we sent the build request to the worker but did not hear back and receive the result.
                 foreach (PipCompletionTask pipCompletionTask in pipCompletionTasks)
                 {
-                    Func<ExecutionResult> executionResultFactory;
-
-                    // The worker goes in a stopped state either by a scheduler request or because it lost connection with the
-                    // remote machine. Check which one applies.
-                    if (IsConnectionLost)
-                    {
-                        // Remember that we have an infrastructure error so we can report it at the end.
-                        // Do not set the flag if the lost connection doesn't affect any pip execution.
-                        m_masterService.HasInfrastructureFailures = true;
-                        executionResultFactory = () =>
-                        {
-                            return LogAndGetNetworkFailureResult(pipCompletionTask, "No result was received and connection was lost");
-                        };
-                    }
-                    else
-                    {
-                        // This one can happen if the user presses CTRL+C
-                        executionResultFactory = () =>
-                        {
-                            var result = new ExecutionResult();
-                            result.SetResult(m_appLoggingContext, PipResultStatus.Canceled);
-                            result.Seal();
-                            return result;
-                        };
-                    }
-
-                    pipCompletionTask.Set(executionResultFactory);
+                    FailRemotePip(pipCompletionTask, "No result was received because connection was lost");

```

Comment:
I'm confused. It looks like we still call FailRemotePip; isn't the idea to re-schedule?

Reply:
The method name is somehow misleading. If you look at the inside of FailRemotePip, we do not set the pip result as failure if the step is MaterializeOutputs. We set the result as NotMaterialized which is treated as Success.

Reply:
> This is a very cool idea and I'm very happy to see it. (Even though, by making BXL more tolerant of ring-machine failures, it will put pressure on cache to become equally resilient.)

The changes are to make "metabuild" error-tolerant. If one of the workers fail to materialize the outputs, the master does not fail. The master would continue the product build with the rest of the workers.